src/main/java/com/yworks/yguard/obf/GuardDB.java

```diff
@@ -423,13 +423,15 @@ else if (STREAM_NAME_MANIFEST.equals(inName.toUpperCase()) ||
             if (size != -1)
             {
 
-              // Create an OutputStream piped through a number of digest generators for the manifest
-              MessageDigest shaDigest = MessageDigest.getInstance("SHA");
-              MessageDigest md5Digest = MessageDigest.getInstance("MD5");
-              DataOutputStream dataOutputStream =
-              new DataOutputStream(new DigestOutputStream(new DigestOutputStream(baos,
-              shaDigest),
-              md5Digest));
+              if (digestStrings == null){
+                digestStrings = new String[]{"SHA", "MD5"};
+              }
+              DataOutputStream dataOutputStream = new DataOutputStream(baos);
+              for (String digestAlg : digestStrings) {
+                // Create an OutputStream piped through a number of digest generators for the manifest
+                MessageDigest digest = MessageDigest.getInstance(digestAlg);
+                dataOutputStream = new DataOutputStream(new DigestOutputStream(dataOutputStream, digest));
```

Comment:
This will result in DaOS(DiOS(DaOS(DiOS(DaOS(baos))
I think keeping it DaOS(DiOS(DiOS(baos) would be better, probably not from a function a and performence perspective, but less confusing.
Also @thomasbehr - do we allow the "modern" for-each in the code?

Reply:
Done:
1. extract fill digests in private method 
2. refactor output streams
3. change SHA algorithm name on SHA-1 as default for resources(forgot to fix in previous commit)

Reply:
Thank you very much for adjusting your code. I think it is fine now and have merged your contribution into master.

Reply:
You are welcome. It was simple task :)